docs/shared/deprecated/as-provided-vs-derived.md

```diff
@@ -0,0 +1,114 @@
+- We converted the provided names to kebab-case. With `as-provided`, we don't transform what's provided.
+
+## Prompting
+
+Until Nx 18, you will be prompted when running a generator to choose which behavior you want to use:
+
+```{% command="nx g lib my-lib --directory=shared/my-lib" path="~/myorg" %}
+✔ Which generator would you like to use? · @nx/node:library
+
+>  NX  Generating @nx/node:library
+
+? What should be the project name and where should it be generated? …
+❯ As provided:
+    Name: my-lib
+    Root: shared/my-lib
+  Derived:
+    Name: shared-my-lib-my-lib
+    Root: packages/shared/my-lib/my-lib
+```
+
+There are two different kinds of generators that will prompt you to choose between `as-provided` and `derived`:
+
+- Project Generators (applications and libraries)
+- Other Code Generators (components and other code)
+
+## Project Generators
+
+As of Nx 16.8.0, generating a project will prompt you to choose how Nx will calculate where the project should be located. You can choose between `as-provided` and `derived`. `as-provided` will be the default in Nx 18.
+
+### `as-provided`
+
+This setting makes app or lib generators behave in the following way:
+
+- `nx g app my-app` creates a new application named `my-app` in the `/my-app` folder
+- `nx g lib my-lib` creates a new library named `my-lib` in the `/my-lib` folder
+- `cd apps/nested/my-app && nx g app my-app` creates a new application named `my-app` in the `/apps/nested/my-app` folder
+- `nx g app my-app --directory=apps/nested/my-app` creates a new application named `my-app` in the `/apps/nested/my-app` folder
+- `nx g lib my-lib --directory=libs/shared/ui/my-lib` creates a new library named `my-lib` in the `/libs/shared/ui/my-lib` folder
+
+### `derived`
+
+Choosing `derived` makes Nx behave the way it did before version 16.8.0. Nx will check the `workspaceLayout` property in `nx.json` to determine how it should calculate the path to the new project.
+
+```json
+{
+  "workspaceLayout": {
+    "appsDir": "demos",
+    "libsDir": "packages"
+  }
+}
+```
+
+These settings would store apps in `/demos/` and libraries in `/packages/`. The paths specified are relative to the
+workspace root.
+
+This makes app or lib generators behave in the following way:
+
+- `nx g app my-app` creates a new application named `my-app` in the `/demos/my-app` folder
+- `nx g lib my-lib` creates a new library named `my-lib` in the `/packages/my-lib` folder
+- `nx g app my-app --directory=nested` creates a new application named `nested-my-app` in the `/demos/nested/my-app` folder
+- `nx g lib my-lib --directory=shared/ui` creates a new library named `shared-ui-my-lib` in the `/packages/shared/ui/my-lib` folder
+
+If you accidentally generate a project in the wrong folder, use the [move generator](/nx-api/workspace/generators/move) to move it to the correct location.
```

Comment:
I don't know if it is worth rehashing all of this here. If you know, you know. If you don't know... well you probably shouldn't learn now because it's about to be removed.

I would leave this section as "This is how Nx used to work. It will calculate. If you are a new user, don't worry about it because it's the old way which will be removed." said in a more elegant way :sweat_smile:

Reply:
All the deprecated docs describe the old way features worked.  New users shouldn't be coming to deprecated doc - they're not the audience.

There are two types of people that come to a deprecated docs page:
1. Someone who realizes the feature they want isn't there any more and want to find out why
2. Someone who is stuck on an old version of Nx and wants information that is accurate for their version

This documentation needs to be here for person #2.